workflow/rules/diffexp.smk

```diff
@@ -262,6 +242,26 @@ rule logcount_matrix:
         "../scripts/sleuth-to-matrix.R"
 
 
+rule plot_diffexp_heatmap:
+    input:
+        logcountmatrix_file="results/tables/logcount-matrix/{model}.logcount-matrix.tsv",
+    output:
+        diffexp_heatmap=report(
+            "results/plots/diffexp-heatmap/{model}.diffexp-heatmap.pdf",
+            caption="../report/plot-heatmap.rst",
+            category="Heatmaps",
+        ),
+    params:
+        model=get_model,
+        predef_genelist=config["diffexp"]["genes_of_interest"],
```

Comment:
This has to be an input file, not a param. Also, remember that this can be missing in the config. Hence you need to put it into a input function which returns `[]` if no genes_of_interest are defined in the config.

Reply:
Noted. Added the function in `common.smk` file.